factory/base.py

```diff
@@ -157,6 +157,14 @@ def __init__(self):
         self.parameters = {}
         self.parameters_dependencies = {}
 
+    @property
+    def sorted_postgen_declarations(self):
+        """Get sorted postgen declaration items."""
+        return sorted(
+            self.postgen_declarations.items(),
+            key=lambda item: item[1].creation_counter,
```

Comment:
When iterating over tuples, I find it easier to read when unused arguments are named, e.g:
```python
return sorted(
    self.postgen_declarations.items(),
    key=lambda _name, declaration: declaration.creation_counter,
)
```

This way, we don't have to refer to the definition of ``postgen_declarations`` to understand its structure.